packages/pluggableWidgets/badge-web/src/Badge.tsx

```diff
@@ -15,10 +12,8 @@ export default function Badge(props: BadgeContainerProps): ReactNode {
     return (
         <DisplayBadge
             type={props.type}
-            value={props.value && props.value.status === ValueStatus.Available ? props.value.value : ""}
-            brandStyle={props.brandStyle}
-            clickable={props.onClick && props.onClick.canExecute}
-            onClick={onClick}
+            value={props.value && props.value.value !== undefined ? props.value.value : ""}
+            onClick={props.onClick && props.onClick.canExecute ? onClick : undefined}
```

Comment:
You might inline this sentence, otherwise the callback is still being created but not used in line 8

Reply:
I dived a bit into the topic. After reading https://reactjs.org/docs/hooks-rules.html, I think we should stick to the rule. It just leads to bugs if you conditionally define hooks on runs. We should conditionally pass a hook to the next component. That’s just how the hook architecture is build (propbably performance is not a big thing). They even have a eslint rule for it. I think we should introduce that rule in our mono repo.